default/python/universe_generation/starsystems.py

```diff
@@ -70,6 +70,16 @@ def name_planets(system):
         fo.set_name(planet, name)
 
 
+def can_have_no_planets(star_type):
+    """
+    Return True is system is allowed to have no planets.
+    This intentionally only checks the STAR_TYPE_MOD_TO_PLANET_SIZE_DIST so that
+    the following code forces most stars to have a planetary body so that in the
+    GUI the presence of a star implies a planet
+    """
+    return universe_tables.STAR_TYPE_MOD_TO_PLANET_SIZE_DIST[star_type][fo.planetSize.noWorld] != 0
```

Comment:
I mean that in the universe_table (source material), if someone sets the values so that 5% of the neutron stars have no planets, then the code (engine) should not have another secret threshold that prevents them from achieving their goal.  

Five percent should mean five percent not zero.

Change the universe_table to reflect the result that we want.